### PR TITLE
deptypes: support special deptypes by string

### PR DESCRIPTION
**Pull request by mathstuf**

### CODE REVIEW

**Comment by adamjstewart** [comment]
Thanks! I think supporting both makes it less confusing. That way you don't have to remember whether to use a string or a magic variable that comes from `from spack import *`.

File: lib/spack/spack/spec.py [hunk 1]
```diff
@@ -155,6 +155,10 @@
 # Special types of dependencies.
 alldeps = ('build', 'link', 'run')
 nolink = ('build', 'run')
+special_types = {
+    'alldeps': ('build', 'link', 'run'),
+    'nolink': ('build', 'run'),
+}
```
Comment (alalazo) [inline lib/spack/spack/spec.py]: @mathstuf Minor thing. What about : 
```python
special_types = {
    'alldeps': alldeps,
    'nolink': nolink,
}
```
if we are going to maintain both ways of referring to special deptypes ?

**Comment by mathstuf** [comment]
Ah, yeah, that makes a lot of sense :) .

**Comment by adamjstewart** [comment]
@tgamblin Can you rebuild the Spack documentation online? It doesn't have anything related to deptypes yet.

**Pull request merged**